.github/workflows/push.yml

```diff
@@ -1,103 +1,224 @@
 name: master
 on: push
 jobs:
+  prepare-cache:
+    runs-on: ${{ matrix.os }
+    strategy:
+      matrix:
+        os: ["ubuntu-latest",  "windows-latest"]
+        python-version: ["3.6", "3.7", "3.8"]
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 1
+      - name: Cache conda
+        uses: actions/cache@v1
+        with:
+          path: ~/conda_pkgs_dir
+          key: ${{ matrix.os }}-python-${{ matrix.python-version }}-conda-${{ github.run_id }}-${{ hashFiles('environment.yml') }}
+      - name: Cache pip if Ubuntu
+        if: ${{ matrix.os == 'ubuntu-latest' }}
+        uses: actions/cache@v1
+        with:
+          path: ~/.cache/pip
+          key: ${{ runner.os }}-python-${{ matrix.python-version }}-pip-${{ github.run_id }}-${{ hashFiles('environment.yml') }}
+      - name: Cache pip if Windows
+        if: ${{ matrix.os == 'windows-latest' }}
```

Comment:
again, why only pip is handled differently on Linux vs Windows? also, where's MacOS specific section?

Reply:
I will see. We do not test on MacOS.

Reply:
I believe, I answered.